src/Microsoft.OpenApi.Readers/V3/OpenApiV3VersionService.cs

```diff
@@ -129,6 +129,20 @@ public OpenApiV3VersionService(OpenApiDiagnostic diagnostic)
                         }
                         id = localSegments[3];
                     }
+                    else if (id.StartsWith("/paths/"))
+                    {
+                        var localSegments = segments[1].Split('/');
```

Comment:
Getting a compile error when I add this, the error:
```
Microsoft.OpenApi.Readers\V3\OpenApiV3VersionService.cs(134,68,134,105): error CS1503: Argument 2: cannot convert from 'System.StringSplitOptions' to 'char'
```

I checked the [documentation ](https://learn.microsoft.com/en-us/dotnet/api/system.string.split?view=net-7.0#system-string-split(system-char-system-stringsplitoptions))and this is defined
```
public string[] Split (char separator, StringSplitOptions options = System.StringSplitOptions.None);
```

But when I check the code in System.String there is no such method only
```
public String[] Split(char[] separator, StringSplitOptions options);
```

so shall have to make the first parameter an array as well in this case.
Further we now have one less segment because the `/paths/....` starts with a `/` the first segment will be an empty string and will be removed, so have to check whether we have 2 segments instead of 3.

Reply:
Looking at the next issue I think the same applies here, that method overload is not defined for the specified target framework netstandard2.0?

Reply:
I can see that you have already moved to an array, can you move the array to a constant so we avoid unnecessary allocations please?

Reply:
I am sorry but I am rather lost here, I have never seen an array being declared as a constant in C#, I do not know how to do that, can you clarify or give an example please. I am not sure also why this is necessary in the first part of the if for `"/components/"` it only does `var localSegments = segments[1].Split('/');` and there it was fine, for the `/paths/` check I did the same, so why is it not OK in this case?

Reply:
by constant I meant `private static readonly char[] Name = new char[] {'/'}'`

Reply:
Thank you for clarifying, I declared a static readonly private variable `_pathSeparator` at the start of the class as you suggested.